src/components/SecondaryPanes/Expressions.js

```diff
@@ -125,9 +140,18 @@ class Expressions extends Component<Props, State> {
   }
 
   handleChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
-    this.setState({ inputValue: e.target.value });
+    const target = e.target;
```

Comment:
Slight improvement here since `findAutocompleteMatches` is a redux action you're debouncing let's do this and give it a slight head start on the react state's update.

```js
this.setState(() => {
  const { target } = e;
  if (features.autocompleteExpression) {
    this.findAutocompleteMatches(target.value, target.selectionStart);
  }
  return { inputValue: target.value }
});
```

Reply:
I tried this solution, but I learned now that a synthetic event is pooled. Can I write `e.persist()`? I read that the event is pooled for performance reasons and I'm not sure that I can do it.

Reply:
Hmm yeah let's avoid the event pooling for now. We could still put the call to `findAutocompleteMatches` before `setState`. Thanks for following up on that.